File: app/controllers/Accounts.scala
package controllers

import models._
import play.api.libs.json._
import play.api.mvc._


object Accounts extends Controller with Security {

  def findAll() = HasToken() { _ => currentUserId => implicit request =>
    Ok(Json.toJson(
        Account.forUser(currentUserId)
      ))
  }

  def findOne(id: Long) = HasToken() { _ => currentUserId => implicit request =>
    Account.find(id) match {
      case Some(acc) => Ok(Json.toJson(acc))
      case None => NotFound(s"Account with id '$id not found")
    }
  }

  def save() = HasToken() { _ => currentUserId => implicit req =>
    Account.form.bindFromRequest.fold(
      errors => BadRequest(errors.errorsAsJson),
      account =>
        account.saveEither match {
          case Right(acc) => Ok(Json.toJson(account))
          case Left(errors) => BadRequest(errors.toString())
        }
    )
  }

  def remove(id: Long) = HasToken() { _ => currentUserId => implicit req =>
    Account.find(id) match {
      case Some(acc) =>
        acc.delete()
        Ok(s"Account with id '$id removed")
      case None => NotFound
    }
  }

}

File: app/assets/src/accounts/accounts.js
angular.module("docstore.accounts", [])
    .factory("Accounts", ["$http", function ($http) {
        return {
           /* save: function (data) {
                return $http.post("/api/organizations", data);
            },
            remove: function (id) {
                return $http.delete("/api/organizations/" + id);
            },
            findOne: function (id) {
                return $http.get("/api/organizations/" + id);
            },*/
            findAll: function () {
                return $http.get("/api/accounts");
            }
        };
    }])
;